plugins/alexa/__init__.py

```diff
@@ -0,0 +1,127 @@
+
+#!/usr/bin/env python3
+# vim: set encoding=utf-8 tabstop=4 softtabstop=4 shiftwidth=4 expandtab
+#########################################################################
+#  Copyright 2016 Kai Meder <kai@meder.info>
+#########################################################################
+#  This file is part of SmartHomeNG.
+#
+#  SmartHomeNG is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  SmartHomeNG is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with SmartHomeNG. If not, see <http://www.gnu.org/licenses/>.
+#########################################################################
+
+import logging
+import socket
+import json
+from lib.model.smartplugin import SmartPlugin
+
+from .device import AlexaDevices, AlexaDevice
+from .action import AlexaActions
+from .service import AlexaService
+
+from . import actions_turn
+from . import actions_temperature
+from . import actions_percentage
+
+class Alexa(SmartPlugin):
+    PLUGIN_VERSION = "0.7.0"
+    ALLOW_MULTIINSTANCE = False
+
+    def __init__(self, sh, service_host='0.0.0.0', service_port=9000):
+        self.logger = logging.getLogger(__name__)
+        self.sh = sh
+        self.devices = AlexaDevices()
+        self.actions = AlexaActions(self.sh, self.logger, self.devices)
+        self.service = AlexaService(self.sh, self.logger, self.PLUGIN_VERSION, self.devices, self.actions, service_host, service_port)
+
+    def run(self):
+        self.validate_devices()
+        self.service.start()
+        self.alive = True
+
+    def stop(self):
+        self.service.stop()
+        self.alive = False
+
+    def parse_item(self, item):
+        # device/appliance
+        device_id = None
+        if 'alexa_device' in item.conf:
```

Comment:
You should use `self.has_iattr()` and `self.get_iattr_value()` instead of direct access to `item.conf`.